spec/services/preserved_object_handler_check_exist_spec.rb

```diff
@@ -0,0 +1,756 @@
+        let(:updated_status_msg_regex) { Regexp.new(Regexp.escape("#{exp_msg_prefix} PreservedCopy status changed from")) }
+
+        it 'calls Stanford::StorageObjectValidator.validation_errors for moab' do
+          mock_sov = instance_double(Stanford::StorageObjectValidator)
+          expect(mock_sov).to receive(:validation_errors).and_return([])
+          allow(Stanford::StorageObjectValidator).to receive(:new).and_return(mock_sov)
+          po_handler.check_existence
+        end
+
+        context 'when moab is valid' do
+          context 'PreservedCopy' do
+            context 'changed' do
+              before do
+                allow(po_handler).to receive(:moab_validation_errors).and_return([])
+              end
+              it 'version to incoming_version' do
+                orig = pc.version
+                po_handler.check_existence
+                expect(pc.reload.version).to be > orig
+                expect(pc.reload.version).to eq incoming_version
+              end
+              it 'size if supplied' do
+                orig = pc.size
+                po_handler.check_existence
+                expect(pc.reload.size).not_to eq orig
+                expect(pc.reload.size).to eq incoming_size
+              end
+              it 'last_audited' do
+                orig = Time.current.to_i
+                pc.last_audited = orig
+                pc.save!
+                sleep 1 # last_audited is bigint, and granularity is second, not fraction thereof
+                po_handler.check_existence
+                expect(pc.reload.last_audited).to be > orig
+              end
+              it 'last_checked_on_storage' do
+                orig = Time.current
+                pc.last_checked_on_storage = orig
+                pc.save!
+                po_handler.check_existence
+                expect(pc.reload.last_checked_on_storage).to be > orig
+              end
+              it 'updated_at' do
+                orig = pc.updated_at
+                po_handler.check_existence
+                expect(pc.reload.updated_at).to be > orig
+              end
+              it 'status becomes "ok" if it was invalid_moab (b/c after validation)' do
+                pc.status = PreservedCopy::INVALID_MOAB_STATUS
+                pc.save!
+                po_handler.check_existence
+                expect(pc.reload.status).to eq PreservedCopy::OK_STATUS
+              end
+            end
+            context 'unchanged' do
+              before do
+                allow(po_handler).to receive(:moab_validation_errors).and_return([])
+              end
+              it 'status if former status was ok' do
+                pc.status = PreservedCopy::OK_STATUS
+                pc.save!
+                po_handler.check_existence
+                expect(pc.reload.status).to eq PreservedCopy::OK_STATUS
+              end
+              it 'size if incoming size is nil' do
```

Comment:
looked at this with @tallenaz, and we think this is as it should be.  it appears that we test non-nil size above:  https://github.com/sul-dlss/preservation_catalog/pull/423/files/0c173915c5ecc6045581a9fcc72104306344b640..361d44c0385f5b9e8539fa73fb7cf8031297e806#diff-a86addf91eaa9d589fbe0483306dc0ccR139

and that the tests are grouped by "situations where fields should change" and "situations where fields should remain unchanged", since the overall context is that the incoming version is greater than the catalog version and that the moab is valid, which indicates that the preserved copy should change in general (at least its `version` and `updated_at` fields, which are notably absent from the `context 'unchanged'` section).